apps/hash-ai-worker-py/app/_status.py

```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```

Comment:
This confuses me as `mypy` itself uses this annotation

Reply:
https://github.com/pydantic/pydantic/issues/2678#issuecomment-1008139014 (2021) about the view from pydantic.

Here is a real-life example that would not get caught:

```python
from __future__ import annotations
from typing if TYPE_CHECKING

if TYPE_CHECKING:
	from .models import Example

class A(BaseModel):
	example: Example
```

looks correct, right? but pydantic will fail, this is the same as:

```python
class A(BaseModel):
	example: "Example"
```

`"Example"` effectively turns into an implicit `ForwardRef` with no reference to `Example`, `A` does not know what `Example` is referring to, and the construction of any object will fail because there's no distinction between explicit forward refs (via quoted strings) and real imports A) pydantics work is much hard, B) to debug these issues is _a pain_.

The reason why `mypy` uses this annotation is that it relies on static analysis (instead of additional runtime analysis guarantees that beartype and pydantic provide), so it is "fine" for them to be strings as they don't need the actual annotation information you can retrieve at runtime.

Reply:
Ah, got it, thanks for clarifying 👍